sql-plugin/src/main/scala/org/apache/spark/sql/hive/rapids/GpuHiveTableScanExec.scala

```diff
@@ -535,4 +536,63 @@ class GpuHiveDelimitedTextPartitionReader(conf: Configuration,
       }
     }
   }
+
+  /**
+   * Override of [[com.nvidia.spark.rapids.GpuTextBasedPartitionReader.castStringToDate()]],
+   * to convert parsed string columns to Dates.
+   * Two key differences from the base implementation, to comply with Hive LazySimpleSerDe
+   * semantics:
+   *   1. The input strings are not trimmed of whitespace.
+   *   2. Invalid date strings do not cause exceptions.
+   */
+  override def castStringToDate(input: ColumnVector, dt: DType): ColumnVector = {
+    val cudfFormat = DateUtils.toStrf(dateFormat, parseString = true)
+    withResource(input.isTimestamp(cudfFormat)) { isDate =>
+      withResource(input.asTimestamp(dt, cudfFormat)) { asDate =>
+        withResource(Scalar.fromNull(dt)) { nullScalar =>
+          isDate.ifElse(asDate, nullScalar)
+        }
+      }
+    }
+  }
+
+  // As per Hive's LazySimpleSerDe documentation, only the following timestamp pattern is supported:
+  //  "uuuu-MM-dd HH:mm:ss[.SSS...]"
+  //
+  // Note: No support for "uuuu-MM-dd'T'HH:mm:ss[.SSS...][Z]"
+  override def timestampFormat: String = s"${DateFormatter.defaultPattern} HH:mm:ss[.SSS][XXX]"
```

Comment:
This is never used???? Should we use it? should we comment about it, or make a change in some way that we don't set it?  It just feels odd that we need set it, but we don't use it.

Reply:
Ah, this isn't used in the current iteration. It used to be used earlier, when deriving the filtering regex.